builtin/providers/aws/resource_aws_redshift_cluster.go

```diff
@@ -0,0 +1,574 @@
+		ClusterVersion:      aws.String(d.Get("cluster_version").(string)),
+		NodeType:            aws.String(d.Get("node_type").(string)),
+		DBName:              aws.String(d.Get("database_name").(string)),
+		AllowVersionUpgrade: aws.Bool(d.Get("allow_version_upgrade").(bool)),
+	}
+	if d.Get("cluster_type") == "multi-node" {
+		createOpts.NumberOfNodes = aws.Int64(int64(d.Get("number_of_nodes").(int)))
+	}
+	if v := d.Get("cluster_security_groups").(*schema.Set); v.Len() > 0 {
+		createOpts.ClusterSecurityGroups = expandStringList(v.List())
+	}
+
+	if v := d.Get("vpc_security_group_ids").(*schema.Set); v.Len() > 0 {
+		createOpts.VpcSecurityGroupIds = expandStringList(v.List())
+	}
+
+	if v, ok := d.GetOk("cluster_subnet_group_name"); ok {
+		createOpts.ClusterSubnetGroupName = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("availability_zone"); ok {
+		createOpts.AvailabilityZone = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("preferred_maintenance_window"); ok {
+		createOpts.PreferredMaintenanceWindow = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("cluster_parameter_group_name"); ok {
+		createOpts.ClusterParameterGroupName = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("automated_snapshot_retention_period"); ok {
+		createOpts.AutomatedSnapshotRetentionPeriod = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("publicly_accessible"); ok {
+		createOpts.PubliclyAccessible = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("encrypted"); ok {
+		createOpts.Encrypted = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("elastic_ip"); ok {
+		createOpts.ElasticIp = aws.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] Redshift Cluster create options: %s", createOpts)
+	resp, err := conn.CreateCluster(createOpts)
+	if err != nil {
+		log.Printf("[ERROR] Error creating Redshift Cluster: %s", err)
+		return err
+	}
+
+	log.Printf("[DEBUG]: Cluster create response: %s", resp)
+	d.SetId(*resp.Cluster.ClusterIdentifier)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"creating", "backing-up", "modifying"},
+		Target:     "available",
+		Refresh:    resourceAwsRedshiftClusterStateRefreshFunc(d, meta),
+		Timeout:    5 * time.Minute,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("[WARN] Error waiting for Redshift Cluster state to be \"available\": %s", err)
+	}
+
+	return resourceAwsRedshiftClusterRead(d, meta)
+}
+
+func resourceAwsRedshiftClusterRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).redshiftconn
+
+	log.Printf("[INFO] Reading Redshift Cluster Information: %s", d.Id())
+	resp, err := conn.DescribeClusters(&redshift.DescribeClustersInput{
+		ClusterIdentifier: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			if "ClusterNotFound" == awsErr.Code() {
+				d.SetId("")
+				log.Printf("[DEBUG] Redshift Cluster (%s) not found", d.Id())
```

Comment:
need this log statement before you call `d.SetId("")`, otherwise you'll log an `""` :smile: